addons/hr/tests/test_self_user_access.py

```diff
@@ -209,10 +209,9 @@ def testWriteOtherUserPreferencesEmployee(self):
                 self.hubert.with_user(self.richard).write(v)
 
     def testWriteSelfPhoneEmployee(self):
-        # phone is a related from res.partner (from base) but added in SELF_READABLE_FIELDS
         self.env['ir.config_parameter'].set_param('hr.hr_employee_self_edit', False)
         with self.assertRaises(AccessError):
-            self.richard.with_user(self.richard).write({'phone': '2154545'})
+            self.richard.with_user(self.richard).write({'work_phone': '2154545'})
```

Comment:
Hello @ibo-odoo  :wave: ,
It seems this test was to check the phone field, and should raise an access error from orm.
But here using `work_phone` raises a different access error from res.user(inherit in hr). Something like restricting hr fields
And this type of access error is already handled in `testWriteSelfUserEmployeeSettingFalse`. 

Maybe removing this test is better.

Thanks

Reply:
I think this is not the case that `this type of access error is already handled in testWriteSelfUserEmployeeSettingFalse`, the test case is specific to the `hr_employee_self_edit` functionality. 

However, as we discussed, earlier the phone field existed in the res_users model in the HR module, and it was later replaced by the employee_phone field. After that, this field was removed as well. It's possible that this test case is no longer needed. Therefore, removing it.
(Leaving this conversation open for feedback from other validators.)